### PR TITLE
feat: Complete predicate builders for all operators.

### PR DESCRIPTION
**Pull request by QuakeWang**
fix: #266

### CODE REVIEW

File: crates/iceberg/src/expr/predicate.rs [hunk 1]
```diff
@@ -282,15 +282,18 @@ impl Bind for Predicate {
...
-                        if !bound_expr.term.field().field_type.is_floating_type() {
-                            return Err(Error::new(
-                                ErrorKind::DataInvalid,
-                                format!(
-                                    "Expecting floating point type, but found {}",
-                                    bound_expr.term.field().field_type
-                                ),
-                            ));
+                    &PredicateOperator::IsNan => {
+                        if bound_expr.term.field().field_type.is_floating_type()
```
Comment (liurenjie1024) [inline crates/iceberg/src/expr/predicate.rs:286]: Sorry, I don't quite the point of this rewriting. Why if `a` is floating or required, `a is nan` is always false?

File: crates/iceberg/src/expr/predicate.rs [hunk 2]
```diff
@@ -282,15 +282,18 @@ impl Bind for Predicate {
...
+                            | bound_expr.term.field().required
+                        {
+                            return Ok(BoundPredicate::AlwaysFalse);
+                        }
+                    }
+                    &PredicateOperator::NotNan => {
```
Comment (liurenjie1024) [inline crates/iceberg/src/expr/predicate.rs:292]: Dittol

File: crates/iceberg/src/expr/predicate.rs
Comment (liurenjie1024) [inline crates/iceberg/src/expr/predicate.rs:1]: Could we also add some ut for negates?

**Review by liurenjie1024 — COMMENTED** [review]
@QuakeWang Thanks for this pr, it looks great! I've left some small questions.

**Comment by liurenjie1024** [comment]
cc @QuakeWang Are you still working on this?

**Comment by QuakeWang** [comment]
> cc @QuakeWang Are you still working on this?

Sorry, I've been a bit busy lately, I'll keep following up.

**Review by QuakeWang — COMMENTED** [review]
@liurenjie1024 Hi, I have updated it, PTAL 👀

File: crates/iceberg/src/expr/predicate.rs
Comment (liurenjie1024) [reply crates/iceberg/src/expr/predicate.rs:1]: Sorry, I didn't notice that before.

**Review by liurenjie1024 — APPROVED** [review]
Thanks @QuakeWang for adding this, looks great !

**Pull request merged**